### PR TITLE
Run tests with Python 3.12

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

File: .github/workflows/daily.yml [hunk 1]
```diff
@@ -67,6 +67,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
+          # TODO: Use Python 3.12. As of 2024-03-08, several third-party
+          # packages fail to install with Python 3.12.
           python-version: "3.10"
```
Comment (srittau) [inline .github/workflows/daily.yml:72]: We should probably reuse a common workflow for testing stubs, see https://docs.github.com/en/actions/using-workflows/reusing-workflows

**Comment by srittau** [comment]
The primer failure looks unrelated to the Python version to me: https://github.com/python/typeshed/actions/runs/8205495006/job/22442480594?pr=11544#step:5:1109

Maybe some else has an idea? Otherwise I can try downgrading the Python version used for primer as well.

**Comment by JelleZijlstra** [comment]
The setup.py for confluent-kafka (https://github.com/confluentinc/confluent-kafka-python/blob/v2.1.1/setup.py) does rely on distutils, so maybe something changed there.

**Comment by srittau** [comment]
Makes sense. Distutils is also the reason why stubtest fails for some packages.

**Pull request merged**